tensorflow/python/autograph/operators/py_builtins.py

```diff
@@ -384,8 +384,29 @@ def _py_filter(function, iterable):
   return filter(function, iterable)
 
 
+def any_(iterable):
+  if isinstance(iterable, dataset_ops.DatasetV2):
+    return _tf_dataset_any(iterable)
+  return _py_any(iterable)
+
+
+# any() operation is essentially a "if first True element exist".
+# For that it could be translated to `filter(True)` to filter out
+# only `True` element, and then `take(1)`. This works in tf.data
+# as tf.data's filter+take is done in pipeline so it will stop
+# as soon as `take(1)` returns.
+def _tf_dataset_any(iterable):
+  return iterable.filter(
+      lambda x: x).take(1).reduce(
+          constant_op.constant(False, dtype=dtypes.bool),
+          lambda x, y: math_ops.logical_or(x, y))
```

Comment:
Strictly speaking, I don't think we need the `logical_or`, right?

* if the result is `False`, then the dataset is empty and returns the default `False`
* if the result is `True`, then the dataset has exactly one element and that is always `True`, thanks to the preceding `filter`

So this should work (with this briefly explained in a comment):

`reduce(constant_op.constant(False, dtype=dtypes.bool), lambda _, y: y)`

WDYT?

Reply:
Thanks @mdanatg, that makes sense. PR has been updated.